### PR TITLE
Fixed LDAP ObjectSid mapping

### PR DESCRIPTION
**Pull request by cam92yz**
So far we have solved the problem by recompiling it and replacing the jar file. But that is not a solution for the future. I saw that others also have the same problem:

https://keycloak.discourse.group/t/active-directory-user-federation-not-mapping-objectsid-as-expected/9225

Here is our fix. Since I don't know your code, this is kind of a poc. Maybe one of you has a better idea.

Closes #24501

### CODE REVIEW

**Comment by ahus1** [comment]
Thank you for submitting this draft PR. While this looks valid, please create a bug report first detailing the scenario when this is important and why this should change. This helps us to have all the information in GitHub should there be questions about this in the future. 

Once there is an issue, you or a maintainer can link it with this PR. Please comment here once you've created an issue.

**Comment by cam92yz** [comment]
https://github.com/keycloak/keycloak/issues/24501

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPIdentityStore.java [hunk 1]
```diff
@@ -486,8 +488,13 @@ private LDAPObject populateAttributedType(SearchResult searchResult, LDAPQuery l
                     NamingEnumeration<?> enumm = ldapAttribute.getAll();
                     while (enumm.hasMoreElements()) {
                         Object val = enumm.next();
-
-                        if (val instanceof byte[]) { // byte[]
+                        
+                        if(ldapAttributeName.toLowerCase().equals("objectsid"))
```
Comment (rmartinc) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/idm/store/ldap/LDAPIdentityStore.java:492]: As I commented in the issue, I feel that using hardcoded `objectsid` here is not OK. Maybe we should add a specific SID mapper in the msad folder. Similar to what we are doing with `MSADUserAccountControlStorageMapperFactory` but for SID syntax attributes.

**Comment by docholiday** [comment]
Hi there, as far as I understand, the issue with the objectSid has been fixed with this. However, I still don't know what to do to get a correct mapping.
I'm running Keycloak 25.02. And neither can I find another mapper nor is there any option to set.
So what I receive is still something like that: AQUAAAAAAAUVAAAAP82NyY0SOfXPiOjKgxAAAA==

Any help greatly appreciated!

**Comment by rmartinc** [comment]
This PR was closed and the related issue #24501 was moved to an enhancement that has not been done.

**Comment by docholiday** [comment]
Ah, ok. Thanks for the reply!

**Pull request closed without merging**